### PR TITLE
SV Rocket

### PR DESCRIPTION
**Pull request by Lokey82**
## About the PR

This pull request adds my latest creation: The SV Rocket, a small, incredibly cheap, and questionably safe ship that I originally built as a joke hence the name being a play on words referencing the fact that the walls of this ship are made of rock.
With only a price listed of 7000 spessos this makes a great ship for those who find themselves in financial trouble.
Ironically enough as well, this new ship is also built for a variety of tasks (such as smuggling if you have a name hiding IFF, SSH) such as mining, even coming pre-equipped with very cheap and basic mining supplies (A pickaxe, emergency EVA, and mag boots)
A goal I had for creating this ship was also making it if not the cheapest, one of the cheapest ships on the market which meant of course certain budget cuts had to be made to ensure this ship would be as affordable as it is, some of these include: Substituting a gravity generator with mag boots, the generator powering the APC directly rather than using a SMES and substation, Oxygen supply is a single oxygen canister, The lighting is provided by a floodlight (cell charger included), All windows are replaced with inflatable barriers, and it only starts with 5 sheets of plasma for the generator.
All these budget cuts not only make it an incredibly cheap yet still decent choice of ship but also enhances the feel of the ship's shoddiness as it is not only an SV, it is a little hard to imagine a luxury ship with every necessity met crammed into a tiny asteroid shaped pile held together by one's sheer will not to use a pickaxe on it.

Ultimately this pile of rock that I dare call a ship despite being created as a joke I believe should make a wonderful addition to the game for those who are looking for something cheap yet still somewhat reliable. (or a flying asteroid that can easily pass as a real one with the right IFF console.)
## Why / Balance

As mentioned above, this ship will not only be great for those looking for something incredibly cheap but also accidentally made perfect for smuggling as it is an incredibly small ship shaped like an asteroid and I feel with the use of a basic IFF console it can possibly pass as a asteroid as well at first glance.
## Technical details

The only code used in this project was the shipyard file which I seriously hope works.
## Media

Ingame
Ingame view (FOV off)

Mapper
Mapping view

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- add: One of NanoTrasen's ship engineers thought it would be funny if they tried to sell the crew a rock, So we now introduce the all new SV Rocket! (pun intended)

### CODE REVIEW

**Comment by Lokey82** [comment]
By the way: I understand the docks are kitty-cornered, I do have a fix planned that will release tomorrow.

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+- type: vessel
+  id: rocket
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:2]: ``id: Rocket``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  name: SV Rocket
+  description: (pun intended)
+  price: 7000
+  category: Small
+  group: Civilian
+  shuttlePath: /Maps/Shuttles/rocket.yml
+
+- type: gameMap
+  id: rocket
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:11]: ``id: Rocket``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  mapName: 'SV Rocket'
+  mapPath: /Maps/Shuttles/rocket.yml
+  minPlayers: 0
+  stations:
+    rocket:
+      stationProto: StandardFrontierVessel
+      components:
+        - type: StationNameSetup
+          mapNameTemplate: 'SV14 Rocket {1}'
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:20]: ``mapNameTemplate: 'Rocket {1}'``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  group: Civilian
+  shuttlePath: /Maps/Shuttles/rocket.yml
+
+- type: gameMap
+  id: rocket
+  mapName: 'SV Rocket'
+  mapPath: /Maps/Shuttles/rocket.yml
+  minPlayers: 0
+  stations:
+    rocket:
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:16]: ``Rocket:``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  description: (pun intended)
+  price: 7000
+  category: Small
+  group: Civilian
+  shuttlePath: /Maps/Shuttles/rocket.yml
+
+- type: gameMap
+  id: rocket
+  mapName: 'SV Rocket'
+  mapPath: /Maps/Shuttles/rocket.yml
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:13]: Move the ship file and update the line
``/Maps/_NF/Shuttles/rocket.yml``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      components:
+        - type: StationNameSetup
+          mapNameTemplate: 'SV14 Rocket {1}'
+          nameGenerator:
+            !type:NanotrasenNameGenerator
+            prefixCreator: '14'
+        - type: StationJobs
+          overflowJobs: []
+          availableJobs:
+            Passenger: [ 0, 0 ]
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:27]: No jobs on SV
``availableJobs: {}``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 7]
```diff
@@ -0,0 +1,27 @@
+- type: vessel
+  id: rocket
+  name: SV Rocket
+  description: (pun intended)
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:4]: Add actual description, no ``( )``

File: Resources/Prototypes/_NF/Shipyard/rocket.yml [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+  price: 7000
+  category: Small
+  group: Civilian
+  shuttlePath: /Maps/Shuttles/rocket.yml
```
Comment (dvir001) [inline Resources/Prototypes/_NF/Shipyard/rocket.yml:8]: Move the ship file and update the line
``/Maps/_NF/Shuttles/rocket.yml``

File: Resources/Maps/Shuttles/rocket.yml [hunk 9]
```diff
@@ -0,0 +1,828 @@
+meta:
```
Comment (dvir001) [inline Resources/Maps/Shuttles/rocket.yml:1]: Move this file to
``/Maps/_NF/Shuttles/rocket.yml``

**Review by dvir001 — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/_NF/Shipyard/rocket.yml (hunk 5 above)
Comment (Lokey82) [reply Resources/Prototypes/_NF/Shipyard/rocket.yml]: I searched far and wide throughout the repo, I don't believe the shuttle file you want me to move it to exists, And if it does a full file path would be helpful.

Comment (dvir001) [reply Resources/Prototypes/_NF/Shipyard/rocket.yml]: Yes, its not there since you need to move it there.

Comment (Lokey82) [reply Resources/Prototypes/_NF/Shipyard/rocket.yml]: Ah my bad, I just realized that my branch was out of sync with the master, I'm making the movements now.

**Comment by Lokey82** [comment]
It appears that VS is having a bit of trouble pushing my latest commit with all the filepath changes.
Expect a fix and the commit to be pushed hopefully tomorrow.

**Comment by Lokey82** [comment]
After lots of messing around, I finally got it working again! I hope I didn't make too big a mess doing so.

**Comment by Lokey82** [comment]
> image Remove the extra copy of the file, im moving this to draft till its ready for the actual ship review.

Alrighty then! Just let me know what needs to get done and hopefully, it will be back on the review list soon!

**Comment by Lokey82** [comment]
Well, I made some changes, Hopefully, they are satisfactory!

**Comment by MagnusCrowe** [comment]
I don't think this ship should be added.  It's a trap of a ship for players who have very little money.  If checkraze disagrees, he can approve it personally.

**Comment by Cheackraze** [comment]
> I don't think this ship should be added. It's a trap of a ship for players who have very little money. If checkraze disagrees, he can approve it personally.

it can go in the scrapyard. We should put out a call for someone to make a proper scrapyard POI and move the terminal out there, or maybe we can put it on Tinnia's or something.

**Comment by Lokey82** [comment]
> Looks like you have a leftover file in there as well. Clean that up and put it in the scrapyard and then I will merge it

I am happy to hear that my ship might get in, it was indeed intended to be a scrapyard ship as per the fleet designation SV, Sorry about the extra file, that has been plaguing this PR for a little while but I will see if I can finally get rid of it.

**Comment by MagnusCrowe** [comment]
I don't think this ship is a good fit for the server.
I gave you feedback on discord on how to make it something that we would want to have on the server but it seems the advice was ignored.

I am going to close this PR for now.  If we decide that we want to include it at a later date, we can open the PR again.

**Pull request closed without merging**